portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -131,15 +200,45 @@ private int getNumberToDisplay(HttpServletRequest request)
   /** Returns all user comments stored in datastore. */
   private PreparedQuery getAllComments() {
     Query query = new Query("Comment")
-      .addSort("time", SortDirection.DESCENDING);
+      .addSort(EntityProperties.COMMENT_TIMESTAMP, SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     return datastore.prepare(query);
   }
 
+  /** 
+    * Returns all user comments that contain `search` in either
+    * the comment text or author name. 
+    * @param comments List containing all comments in database.
+    * @param search A string to filter the comments by. 
+    */
+  private ArrayList<Comment> getFilteredComments( 
+    ArrayList<Comment> comments, String search) {
+    ArrayList<Comment> filteredComments = new ArrayList<Comment>();
+    Iterables.filter(comments, c -> satisfiesSearch(c, search))
```

Comment:
optional: can try use .stream().filter(...).collect

Reply:
Thanks for the tip! I like this method you've suggested much better than what I was using before. I've changed it!